class.eventcalendar.plugin.php

```diff
@@ -1,265 +1,279 @@
+      $Sender->SetData('CategoriesLabel', 'Please choose categories in which the creation of events should be allowed');
+
+      $Validation = new Gdn_Validation();
+      // $Validation->ApplyRule('Plugins.EventCalendar.CategoryIDs', 'RequiredArray', T('You have to choose at least one category. If you don\'t want to use the plugin any longer, please deactivate it'));
+      $ConfigurationModel = new Gdn_ConfigurationModel($Validation);
+      $ConfigurationModel->SetField(array('Plugins.EventCalendar.CategoryIDs'));
+
+      $Form = $Sender->Form;
+      $Sender->Form->SetModel($ConfigurationModel);
+
+      if ($Sender->Form->AuthenticatedPostBack() != FALSE) {
+         if ($Sender->Form->Save() != FALSE) {
+            $Sender->StatusMessage = T('Saved');
+         }
+      } else {
+         $Sender->Form->SetData($ConfigurationModel->Data);
+      }
+
+      $CategoryModel = new Gdn_Model('Category');
+      $Sender->CategoryData = $CategoryModel->GetWhere(array('AllowDiscussions' => 1, 'CategoryID <>' => -1));
+      $Sender->EventCategory = C('Plugins.EventCalendar.CategoryIDs');
+
+      $Sender->Render('settings', '', 'plugins/EventCalendar');
+   } // End of SettingsController_EventCalendar_Create
+
+
+   /**
+    * Adds menu entry for calendar
+    *
+    * @param object $Sender Base Controller
+    */   
+   public function Base_Render_Before($Sender) {
+      if(CheckPermission('Plugins.EventCalendar.View') && $Sender->Menu) {
+         $Sender->Menu->AddLink(T('Event Calendar'), T('EventCalendar'), 'eventcalendar');
+      }
+   }
+   
+   /**
+    * Adds input fields to new discussion form
+    * Check's for CategoryID, Add and Manage category
+    * Allows creation of events in current and next year
+    * Datefield is prefilled with current date by eventcalendar.js
+    *
+    * @param object $Sender PostController
+    */ 
+   public function PostController_BeforeBodyInput_Handler($Sender) {
+      if(!CheckPermission(array('Plugins.EventCalendar.Add', 'Plugins.EventCalendar.Manage'))) {
+         return;
+      }
+
+      $Sender->AddJsFile('eventcalendar.js', 'plugins/EventCalendar');
+	  $Sender->AddCssFile('eventcalendar.css', 'plugins/EventCalendar');
+	  $Sender->AddCssFile('jquery-ui.css', 'plugins/EventCalendar');	  	
+	  $Sender->AddJsFile('jquery-ui.js', 'plugins/EventCalendar');
+      $Sender->AddDefinition('EventCalendarCategoryIDs', json_encode(C('Plugins.EventCalendar.CategoryIDs')));
+
+      // initially don't hide elements in allowed categories
+      $CategoryID = $Sender->Discussion->CategoryID;
+      if (!in_array($CategoryID, C('Plugins.EventCalendar.CategoryIDs'))) {
+         $Hidden = ' Hidden';   
+      }
+	  
+	  //Removed when switching drop down fields to a date picker
```

Comment:
If you manage to create the pull request so that the files are comparable, you could safely delete instead of commenting out ;)